src/dbnode/storage/index/block.go

```diff
@@ -833,9 +833,24 @@ func (b *block) AddResults(
 			results.Fulfilled().SummaryString(), blockRange.String())
 	}
 
+	var (
+		plCache         = b.opts.PostingsListCache()
+		readThroughOpts = b.opts.ReadThroughSegmentOptions()
+		segments        = results.Segments()
+	)
+	readThroughSegments := make([]segment.Segment, 0, len(segments))
+	for _, seg := range segments {
+		readThroughSeg := seg
+		if _, ok := seg.(segment.MutableSegment); !ok {
+			// Only wrap the immutable segments with a read through cache.
```

Comment:
Honestly, if it's AddResults path, it's fine to wrap the mutable ones too here (we never actually write to them, they're essentially frozen).

I'll also fix this up so this never happens and we always deliver FSTs here soon.

Reply:
Yeah thats fair, I'm a little hesitant to allow wrapping of mutable segments though just for correctness purposes (I know you're gonna do the F.S.T thing but would suck for this to break months/years down the line). I'll do the mutable segment check, but i'll still allow wrapping if its "sealed"

Reply:
Actually spoke offline and settled on not wrapping the mutable segments due to complications it introduces. We're going to delete the concept of a mutable segment soon anyways so this will become a moot point.